python-project-template/.prepare_project.sh

```diff
@@ -1,8 +1,15 @@
 #!/usr/bin/env bash
 
+echo "Checking virtual environment"
+if [ -z "${VIRTUAL_ENV}" ] && [ -z "${CONDA_PREFIX}" ]; then
```

Comment:
I'm a little apprehensive about this, just because I don't know the universe of virtual environment tools. I'm somewhat inclined to make this something that the user has to interact with. Like:

```
>> bash .prepare_project.sh

=== This scrip with install the package in the current virtual environment ===
Is the correct environment activated?
Continue Y/n
```

Reply:
The down side to this is that if we get to the point where we've got a good testing infrastructure, having a forced interaction like this can become a problem.

Reply:
All that being said, if you're confident that the conditional check covers the vast majority of the use cases, then I'm ok with it. I'm just being overly cautious.

Reply:
It is a good thing to be cautious. I have adopted approach `maturin`, wheel build tool uses:
https://github.com/PyO3/maturin/blob/8fb4e35a2fa875253a9ecfa9f91553f5a91c86b6/src/main.rs#L191

It actually also checks for `.venv/pyvenv.cfg` file and python binary in the same directory, but I though it would be too much for our small script.

As a compromise, I would do this:
1. Check these two vars.
2. If None is set, ask user if they are sure about what are they doing, with the default "y/N" prompt (No is default)

Reply:
That seems like a very reasonable compromise. I'm in favor of it.

Reply:
Implemented in the last commit